core/src/impl/Kokkos_Core.cpp

```diff
@@ -270,8 +271,15 @@ void initialize_backends(const InitArguments& args) {
   Impl::ExecSpaceManager::get_instance().initialize_spaces(args);
 }
 
-void initialize_profiling(const InitArguments&) {
-  Kokkos::Profiling::initialize();
+void initialize_profiling(const InitArguments& args) {
+  Kokkos::Profiling::initialize(args.tool_lib);
+  if (args.tool_help) {
+    if (!Kokkos::Profiling::printHelp(args.tool_args)) {
+      std::cerr << "Tool has not provided a help message" << std::endl;
+    }
+    std::exit(EXIT_FAILURE);
```

Comment:
Why do you print a non-zero error code?  If we end up here it is not an error while parsing arguments.

For reference, neither `--help` nor `--kokkos-help` cause program termination.
I think it is reasonable to terminate here but I will observe it is inconsistent with `--kokkos-help`.  I would be open to change the behavior when `--kokkos-help` is passed (not in this PR).  (I would strongly object to `--help` calling `std::exit` because it plays poorly when composing in user code)

Reply:
> Why do you print a non-zero error code?

Not sure what you mean by "print" here. I feel like not exiting when any flavor of "--help" is requested from the command-line is quite non-standard. I honestly can't think of a single time when I would pass "--help" to a program and have it continue running. 

> I would strongly object to --help calling std::exit because it plays poorly when composing in user code

Yeah, I have generally adopted the LLDB methodology for grouping arguments, if there is a `--` somewhere, I assume the arguments up the first `--` are intended for me and everything after is intended for the app, e.g. `timem ls -t` doesn't parse the args even if I have a `-t` option, `timem -e PAPI_TOT_INS -- ls -t` only parses `-e PAPI_TOT_INS` and drops those args + the `--`, `timem -e PAPI_TOT_INS -- lldb -- sleep 2` still only parses `-e PAPI_TOT_INS`, etc.

Reply:
Sorry I meant "call `std::exit` with a non-zero exit code".

Unconditionally exiting on `--help` can pose problem downstream though because we force user code to parse arguments before calling `Kokkos::initialize` so they can print their own help information.  Suppose the code is using another library that also exit on `--help` such as Google Benchmark and life becomes even more complicated.

Reply:
Honestly I'd like to resolve this conversation in the weekly meeting. I agree with Jonathan's take on the subject, but it is a change from what other parts of Kokkos do, so we should discuss as a group

Reply:
Resolved to call `std::exit(EXIT_SUCCESS)` for `--kokkos-tools-help`. `--kokkos-help` behavior is not currently being changed